Mathlib/Topology/Compactness/CompactlyCoherentSpace.lean

```diff
@@ -0,0 +1,336 @@
+* `CompactlyCoherentSpace.CompactCoherentification.instCompactlyCoherentSpace`: The compact
+  coherentification makes any space into a compactly coherent space.
+* `CompactlyCoherentSpace.compactCoherentification_compactlyCoherentSpace_eq_self`: The compact
+  coherentification of a compactly coherent space `X` preserves the topology on `X`.
+* `CompactlyCoherentSpace.continuous_compactCoherentification_of_continuousOn_compact`: If a map
+  `f : X → Y` is continuous on every compact subset of `X` then it is continuous when viewed as a
+  map from `CompactCoherentification X` to `CompactCoherentification Y`.
+
+## References
+* [J. Munkres, *Topology*][Munkres2000]
+* <https://en.wikipedia.org/wiki/Compactly_generated_space>
+-/
+
+noncomputable section
+
+open Set Set.Notation Topology
+
+/-! ### Compactly coherent spaces-/
+
+/-- A space is a compactly coherent space if the topology is generated by the compact sets. -/
+class CompactlyCoherentSpace  (X : Type*) [TopologicalSpace X] : Prop where
+  /-- A space is a compactly coherent space if the topology is generated by the compact sets.-/
+  isCoherentWith : IsCoherentWith (X := X) {K | IsCompact K}
+
+namespace CompactlyCoherentSpace
+
+universe u
+
+variable {X : Type u} [TopologicalSpace X]
+
+/-- A set `A` in a compactly coherent space is open iff for every compact set `K`, the intersection
+`K ∩ A` is open in `K`. -/
+lemma isOpen_iff [CompactlyCoherentSpace X] (A : Set X) :
+    IsOpen A ↔ ∀ K, IsCompact K → IsOpen (K ↓∩ A) :=
+  IsCoherentWith.isOpen_iff isCoherentWith
+
+/-- A set `A` in a compactly coherent space is closed iff for every compact set `K`, the
+intersection `K ∩ A` is closed in `K`. -/
+lemma isClosed_iff [CompactlyCoherentSpace X] (A : Set X) :
+    IsClosed A ↔ ∀ K, IsCompact K → IsClosed (K ↓∩ A) :=
+  IsCoherentWith.isClosed_iff isCoherentWith
+
+/-- If every set `A` is open if for every compact `K` the intersection `K ∩ A` is open in `K`,
+then the space is a compactly coherent space. -/
+lemma of_isOpen (h : ∀ (A : Set X), (∀ K, IsCompact K → IsOpen (K ↓∩ A)) → IsOpen A) :
+    CompactlyCoherentSpace X where
+  isCoherentWith := {isOpen_of_forall_induced := h}
+
+/-- If every set `A` is closed if for every compact `K` the intersection `K ∩ A` is closed in `K`,
+then the space is a compactly coherent space. -/
+lemma of_isClosed (h : ∀ (A : Set X), (∀ K, IsCompact K → IsClosed (K ↓∩ A)) → IsClosed A) :
+    CompactlyCoherentSpace X where
+  isCoherentWith := IsCoherentWith.of_isClosed h
+
+/-- Every weakly locally compact space is a compactly coherent space. -/
+instance of_weaklyLocallyCompactSpace [WeaklyLocallyCompactSpace X] : CompactlyCoherentSpace X where
+  isCoherentWith := IsCoherentWith.isCompact_of_weaklyLocallyCompact
+
+/-- Every sequential space is a compactly coherent space. -/
+instance of_sequentialSpace [SequentialSpace X] : CompactlyCoherentSpace X where
+  isCoherentWith := IsCoherentWith.isCompact_of_seq
```

Comment:
Could you instead deprecate the existing results? Or, if that is too much work, at least add a TODO somewhere saying that every lemma using `IsCoherentWith` with compact subspaces should b updated to use `CompactlyCoherentSpace`?